vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
Hmm. if `isRecognisedError` is true (i.e. something like an invalid plugin handle was in the request), or if `err` is an empty string (there was no error during parsing, we have a successful `RequestOrResponse`), I want to return the `RequestOrResponse` object in order to progress to writing output. That is what happens here. 

This came about because the tests were not passing for the `json_to_json` case where a known error response is returned from the server (and the end-to-end test using capnp conversion did this), an exception was being thrown. 

Unless I am confused, I think you might be reading the condition as `isRecognisedError || err != ""`